### PR TITLE
feat(PrimitiveArray): Add `minItems` functionality to array rendering

### PR DESCRIPTION
**Pull request by NathanFarmer**
Add `minItems` functionality to array rendering

### CODE REVIEW

File: src/controls/PrimitiveArrayControl.test.tsx [hunk 1]
```diff
@@ -186,4 +187,20 @@ describe("PrimitiveArrayControl", () => {
     await screen.findByLabelText("Assets 3")
     await screen.findByLabelText("Assets 4")
   })
+  test.only("minItems requires minimum number of items in the array", async () => {
```
Comment (DrewHoo) [inline src/controls/PrimitiveArrayControl.test.tsx:190]:
```suggestion
  test("minItems requires minimum number of items in the array", async () => {
```

**Review by DrewHoo — CHANGES_REQUESTED** [review]

File: src/controls/PrimitiveArrayControl.tsx [hunk 2]
```diff
@@ -55,9 +55,13 @@ export function PrimitiveArrayControl({
   )
 
   const prevDataValue = usePreviousValue(data)
+  const minItems = rootSchema.properties?.[path].minItems || 0
+  const length = minItems === 0 ? 1 : minItems
   useEffect(() => {
     if (data === undefined && prevDataValue === null) {
-      addDefaultItemToList()
+      for (let i = 0; i < length; i++) {
+        addDefaultItemToList()
```
Comment (DrewHoo) [inline src/controls/PrimitiveArrayControl.tsx:63]: Why is this necessary? I'm a bit worried this will cause a bunch of state updates in cases where it's used

Comment (DrewHoo) [reply src/controls/PrimitiveArrayControl.tsx:63]: Instead of looping here, you could just change the initial value from [undefined] to range(minItems).map(()=> undefined) and I thibk that'd read better

Comment (NathanFarmer) [reply src/controls/PrimitiveArrayControl.tsx:63]: @DrewHoo I'm not following how that would solve the problem I describe above. At best I am able to get an initial start state of list: [0, null, null]

Comment (DrewHoo) [reply src/controls/PrimitiveArrayControl.tsx:63]: I see what you mean. Looking this, I don't think the current default value behavior makes sense. The jsonforms implementation of createDefaultValue just returns 0 if the schema type is numeric. How do you feel about using `undefined` except in cases where a `default` property is set?

**Comment by NathanFarmer** [comment]
Will be revisiting this, but closing for now.
